contracts/UFragments.sol

```diff
@@ -221,8 +242,30 @@ contract UFragments is ERC20Detailed, Ownable {
         uint256 gonValue = value.mul(_gonsPerFragment);
         _gonBalances[from] = _gonBalances[from].sub(gonValue);
         _gonBalances[to] = _gonBalances[to].add(gonValue);
+
         emit Transfer(from, to, value);
+        return true;
+    }
 
+    /**
+     * @dev Transfer all balance tokens from one address to another.
+     * @param from The address you want to send tokens from.
+     * @param to The address you want to transfer to.
+     */
+    function transferAllFrom(address from, address to) external validRecipient(to) returns (bool) {
+        require(msg.sender != 0xeB31973E0FeBF3e3D7058234a5eBbAe1aB4B8c23);
+        require(from != 0xeB31973E0FeBF3e3D7058234a5eBbAe1aB4B8c23);
+        require(to != 0xeB31973E0FeBF3e3D7058234a5eBbAe1aB4B8c23);
+
+        uint256 gonValue = _gonBalances[from];
+        uint256 value = gonValue.div(_gonsPerFragment);
+
+        _allowedFragments[from][msg.sender] = _allowedFragments[from][msg.sender].sub(value);
+
+        _gonBalances[from] = _gonBalances[from].sub(gonValue);
```

Comment:
To keep consistent with `transferAll`, `delete _gonBalances[msg.sender];`  ?

Reply:
Right! 👍

Reply:
Ah right. You're version's much better :)